Sources/eppo/EppoClient.swift

```diff
@@ -47,52 +48,66 @@ public class EppoClient {
     private let state = EppoClientState()
     
     private init(
-        apiKey: String,
-        host: String = "https://fscdn.eppo.cloud",
+        sdkKey: String,
+        host: String,
         assignmentLogger: AssignmentLogger? = nil,
         assignmentCache: AssignmentCache? = InMemoryAssignmentCache()
     ) {
-        self.apiKey = apiKey
+        self.sdkKey = sdkKey
         self.host = host
         self.assignmentLogger = assignmentLogger
         self.assignmentCache = assignmentCache
         
-        let httpClient = NetworkEppoHttpClient(baseURL: host, apiKey: apiKey, sdkName: "sdkName", sdkVersion: "sdkVersion")
+        let httpClient = NetworkEppoHttpClient(baseURL: host, sdkKey: sdkKey, sdkName: "sdkName", sdkVersion: "sdkVersion")
         let configurationRequester = ConfigurationRequester(httpClient: httpClient)
         self.configurationStore = ConfigurationStore(requester: configurationRequester)
     }
     
-    public static func configure(
-        apiKey: String,
+    public static func initialize(
+        sdkKey: String,
         host: String = "https://fscdn.eppo.cloud",
         assignmentLogger: AssignmentLogger? = nil,
         assignmentCache: AssignmentCache? = InMemoryAssignmentCache()
-    ) -> EppoClient {
-        if instance == nil {
-            instance = EppoClient(apiKey: apiKey, host: host, assignmentLogger: assignmentLogger, assignmentCache: assignmentCache)
+    ) async throws -> EppoClient {
+        return try await withCheckedThrowingContinuation { continuation in
+            initializerQueue.async(flags: .barrier) {
+                if sharedInstance == nil {
+                    sharedInstance = EppoClient(
+                        sdkKey: sdkKey,
+                        host: host,
+                        assignmentLogger: assignmentLogger,
+                        assignmentCache: assignmentCache
+                    )
+                    Task {
+                        do {
+                            try await sharedInstance!.loadIfNeeded()
+                            continuation.resume(returning: sharedInstance!)
+                        } catch {
+                            continuation.resume(throwing: error)
+                        }
+                    }
+                } else {
+                    continuation.resume(returning: sharedInstance!)
+                }
+            }
         }
```

Comment:
This is quite complex and I arrived at it when needing only a single thread to access the critical section, hence `barrier`. Inside if the instance is not created, create one. Then start an SDK owned Task which performs the configuration fetching in the background.

Reply:
chatgpt explains it well - https://chatgpt.com/share/550bea2f-86b2-4c75-83d0-8c18cc9325a9